File: go/LeetCode/reverseVowels.go
package main

import "strings"

func main() {
	
}

func reverseVowels(s string) string {
	t := []byte(s)
	left:=0
	right:=len(s)-1
	for left<right {
		for left<right && !strings.Contains("aeiouAEIOU", string(t[left])) {
			left++
		}
		for left<right && !strings.Contains("aeiouAEIOU", string(t[right])) {
			right--
		}
		if left<right {
			t[left],t[right] = t[right], t[left]
			left++
			right--
		}
	}
	return string(t)
}

File: cdd/LeetCode/LeetCode/reverseVowels.cpp
#include <stdio.h>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
	string reverseVowels(string s)
	{
		auto isVowel = [vowels = "aeiouAEIOU"s](char ch){
			return vowels.find(ch) != string::npos;
		};

		int left = 0, right = s.size() - 1;
		while (left<right)
		{
			while (left < right && !isVowel(s[left]) )
			{
				left++;
			}
			while (left<right && !isVowel(s[right]))
			{
				right--;
			}
			if (left<right)
			{
				char temp = s[left];
				s[left] = s[right];
				s[right] = temp;
			}
			left++;
			right--;
		}
		return s;
	}
};